KEDamage.cs

```diff
@@ -34,16 +32,17 @@ override public float DealDamage()
             data.Pierce = CalculateKEAttenuationSimple(data.Pierce,data.Distance,data.FrictionFactor);
 
             // Calculate effects of ERA
+            UpdateTargetERA(data.Pierce);
             data.Pierce = CalculatePostERAPierce(data.Pierce,target.ERAData.KEFractionMultiplier);
 
+            // Armor degradation
+            UpdateTargetArmor(data.Pierce);
+
             // Calculate final damage
-            float finalDamage = Math.Max(0.0f, target.Armor - data.Pierce)*data.HealthKEFactor;
-            return finalDamage;
+            float damage2HP = Math.Max(0.0f, data.Pierce - target.Armor)*data.HealthKEFactor;
```

Comment:
Rename back to `finalDamage` for clarity and conciseness. Also, don't forget to space out your code. :)